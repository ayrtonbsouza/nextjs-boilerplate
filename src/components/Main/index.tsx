import { Wrapper, Logo, Title, Description, Illustration } from './styles';

const Main = () => (
  <Wrapper>
    <Logo
      src="/img/logo.svg"
      alt="Imagem de um atómo e React Avançado escrito ao lado"
    />
    <Title>React Next.js Boilerplate</Title>
    <Description>
      TypeScript, ReactJS, NextJS and Styled Components boilerplate
    </Description>
    <Illustration
      src="/img/hero-illustration.svg"
      alt="Ilustração de um desenvolvedor frente a uma tela com código"
    />
  </Wrapper>
);

export default Main;
