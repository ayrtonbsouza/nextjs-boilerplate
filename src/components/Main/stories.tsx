import Main from './index';

export default {
  title: 'Main',
  component: Main,
} as Meta;

export const Basic: Story = args => <Main {...args} />;
