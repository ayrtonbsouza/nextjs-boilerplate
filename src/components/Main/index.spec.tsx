import { render, screen } from '@testing-library/react';

import Main from './index';

describe('Main Component', () => {
  it('should be able to render the main component', () => {
    const { container } = render(<Main />);
    expect(
      screen.getByRole('heading', { name: /React Next.js Boilerplate/i }),
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be able to render colors correctly', () => {
    const { container } = render(<Main />);
    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b',
    });
  });
});
