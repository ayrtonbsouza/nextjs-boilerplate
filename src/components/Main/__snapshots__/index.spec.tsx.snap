// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component should be able to render the main component 1`] = `
<main
  class="styles__Wrapper-sc-1nn7afu-0 dEiupY"
>
  <img
    alt="Imagem de um atómo e React Avançado escrito ao lado"
    class="styles__Logo-sc-1nn7afu-1 fEmuLP"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-1nn7afu-2 fijqOJ"
  >
    React Next.js Boilerplate
  </h1>
  <h2
    class="styles__Description-sc-1nn7afu-3 kBjhwm"
  >
    TypeScript, ReactJS, NextJS and Styled Components boilerplate
  </h2>
  <img
    alt="Ilustração de um desenvolvedor frente a uma tela com código"
    class="styles__Illustration-sc-1nn7afu-4 eovGMT"
    src="/img/hero-illustration.svg"
  />
</main>
`;
